File: st1009155progp12023/MainWindow.cs
using System.Collections.ObjectModel;
using System.Windows;

namespace st1009155progp12023.st1009155progp12023
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow : Window
    {
        private ObservableCollection<Module> modules = new ObservableCollection<Module>();

        public MainWindow()
        {
            this.InitializeComponent();
            this.moduleDataGrid.ItemsSource = modules;
        }

        private void AddButton_Click(object sender, RoutedEventArgs e)
        {
            var module = new Module()
            {
                Code = this.codeTextBox.Text,
                Name = this.nameTextBox.Text,
                Credits = int.Parse(this.creditsTextBox.Text),
                ClassHoursPerWeek = int.Parse(this.classHoursTextBox.Text)
            };
            modules.Add(module);
        }
    }
}

File: st1009155progp12023/MainWindow.vb
Imports System.Collections.ObjectModel
Imports System.Windows

Namespace st1009155progp12023
    ''' <summary>
    ''' Interaction logic for MainWindow.xaml
    ''' </summary>
    Public Partial Class MainWindow
        Inherits Window
        Private modules As ObservableCollection(Of [Module]) = New ObservableCollection(Of [Module])()

        Public Sub New()
            Me.InitializeComponent()
            Me.moduleDataGrid.ItemsSource = modules
        End Sub

        Private Sub AddButton_Click(ByVal sender As Object, ByVal e As RoutedEventArgs)
            Dim [module] = New [Module] With {
    .Code = Me.codeTextBox.Text,
    .Name = Me.nameTextBox.Text,
    .Credits = Integer.Parse(Me.creditsTextBox.Text),
    .ClassHoursPerWeek = Integer.Parse(Me.classHoursTextBox.Text)
}
            modules.Add([module])
        End Sub
    End Class
End Namespace
